### PR TITLE
reload secrets after delete

### PR DESCRIPTION
**Pull request by minz1027**
- Reload the secrets after delete it

screen shot 2016-09-30 at 3 17 15 pm

### CODE REVIEW

**Comment by d2lam** [comment]
👍  Here is your favorite:

File: app/application/adapter.js [hunk 1]
```diff
@@ -75,7 +75,7 @@ export default DS.RESTAdapter.extend({
    */
   handleResponse(status, headers, payload, requestData) {
     // handle generically when there is an error key in the payload
-    if (payload.error) {
+    if (payload && payload.error) {
```
Comment (petey) [inline app/application/adapter.js]: We should add a test for the adapter receiving no payload.

Comment (minz1027) [inline app/application/adapter.js]: fixed :)

File: tests/unit/application/adapter-test.js [hunk 2]
```diff
@@ -120,3 +120,15 @@ test('it wraps errors', function (assert) {
...
+test('it takes care of empty payload', function (assert) {
+  let adapter = this.subject();
+
+  const requestData = {
+    url: 'http://localhost:8080/v4/pipelines/1234'
+  };
+
+  const payload = adapter.handleResponse(204, {}, null, requestData);
+
+  assert.ok(payload);
```
Comment (petey) [inline tests/unit/application/adapter-test.js]: Whenever I see `assert.ok` on a value, I always wonder if it is actually a boolean being tested.  In this case it is not a boolean. I suppose we really don't care much what the modified payload looks like in this case, but `assert.deepEqual(payload, {});` would be a more accurate test.

Comment (minz1027) [inline tests/unit/application/adapter-test.js]: Thx Petey! Good to know. Fixed already.

**Pull request merged**